### ISSUE TITLE
No error on length mismatch with unpack assignment into list

### ISSUE THREAD

**Issue by JelleZijlstra** [microsoft/pyright#2873]
```python
[a, b, c] = 1, 2
```
produces no errors in current pyright. Changing the LHS to a tuple produces `Expression with type "tuple[Literal[1], Literal[2]]" cannot be assigned to target tuple`, but at runtime it makes no difference whether the LHS is a tuple or a list, so the type checker should reject the list version too.

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.210, which I just published. It will also be included in the next release of pylance.
